### PR TITLE
n-ary functions (2)

### PR DESCRIPTION
**Pull request by avekens**
continuation of #3997:

* ~fprmappr added, ~mapprop shortened
* ~naryfvalelwrdf added
* label change: ~nary... -> ~1ary...
* Additional theorems for binary functions/operations.

.

### CODE REVIEW

**Review by tirix — APPROVED** [review]
Nice!

File: set.mm [hunk 1]
```diff
@@ -758976,6 +758999,18 @@ this concept is not fully developed yet (it is within a mathbox), and it is
       BCGHUQCUCDUDFUMUEUFUGUHUIUJUKULUN $.
   $}
 
+  ${
+    $d N w $.  $d V w $.  $d X w $.
+    $( An n-ary (endo)function on a set ` X ` expressed as a function over the
+       set of words on ` X ` of length ` n ` .  (Contributed by AV,
+       4-Jun-2024.) $)
+    naryfvalelwrdf $p |- ( ( X e. V /\ N e. NN0 ) -> ( F e. ( X NaryF N )
+                           <-> F : { w e. Word X | ( # ` w ) = N } --> X ) ) $=
```
[the remark this answers is not in the document]
Comment (david-a-wheeler) [reply set.mm:759008]: @tirix : `-aryF` is a cool idea! I like it. I don't know what anyone else thinks about it.

**Review by icecream17 — APPROVED** [review]

**Comment by david-a-wheeler** [comment]
What do you think about the `-aryF` idea (above)?

**Comment by icecream17** [comment]
(if you're asking me, it's a great idea)

**Comment by avekens** [comment]
It's actually an interesting idea to call the class -aryF, so that we will have ` ( N-aryF X ) ` for N-ary endofunctions over X.
I wonder, however, if I have to rewrite all proofs manually, or if there is an automatism which does (most of) the work. Meanwhile, I propose to leave the definition unchanged for the moment (and to merge this PR), and to open an issue for the change of the symbol.

**Comment by tirix** [comment]
Globally replacing the math token ` NaryF ` by ` -aryF ` using any text editor should basically do the trick.

**Comment by avekens** [comment]
> Globally replacing the math token `NaryF` by `-aryF` using any text editor should basically do the trick.

but we have to switch the operands: `( X NaryF N )` => `( N -aryF X )`.

**Comment by tirix** [comment]
Oh yes, you're right, it's not that trivial.
No problem for me to merge as-is, I already approved.

**Comment by avekens** [comment]
Since issue #4035 is opened for changing the symbol and order of operands, this PR can be merged now.

**Comment by david-a-wheeler** [comment]
I'm late to the party, but yes, glad it's merged.  We can discuss swapping the order in https://github.com/metamath/set.mm/issues/4035

**Comment by avekens** [comment]
The new HTML representation of nary functions really looks nice:  (1-aryF 𝑋)  
I wonder, however, if somebody gets confused and asks where the symbols 0-aryF, 1-aryF, 2-aryF and N-aryF are defined (and what the meaning of `( C X )` would be)...

**Comment by icecream17** [comment]
I guess a comment could be added to df-naryf explaining that unlike most operators, the display for `-aryF` only has a space on the right hand side. Internally it is still ~ co , i.e. three classes in a row surrounded by parentheses

**Pull request merged**